x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/custom/response/BaseCustomResponseParser.java

```diff
@@ -36,7 +38,7 @@ public InferenceServiceResults parse(HttpResult response) throws IOException {
         }
     }
 
-    protected abstract T transform(Map<String, Object> extractedField);
+    protected abstract InferenceServiceResults transform(Map<String, Object> extractedField);
```

Comment:
The `TextEmbeddingResponseParser` will now return different types of results depending on the embedding type so we need this to be the base results type to cover all them